### ISSUE TITLE
Build wheels for musllinux (PEP 656) platforms

### ISSUE THREAD

**Issue by mwheeler** [falconry/falcon#2145]
This is a small feature request to build wheels for [PEP 656](https://peps.python.org/pep-0656/) `musllinux` platform tags, if your team aren't opposed to the idea.

We're seeing `pip install` for falcon on AArch64 Alpine Linux platforms take upwards of 20 minutes to build the wheel from source (on Github Actions nodes).

**Comment by CaselIT** [comment]
Hi,

thanks for reporting.
This does seem to be taken case by the cibuildwheel action https://github.com/pypa/cibuildwheel that can be used to create the wheels so it may be a theoretically low effort option option to support

**Comment by vytas7** [comment]
I agree with @CaselIT that https://github.com/pypa/cibuildwheel looks very promising, we should look into it.
@mwheeler just confirm, do you know what architectures are we expected to add `musllinux` support for, do we need the likes of `s390x`, or just x86 and ARM?

**Comment by vytas7** [comment]
FWIW, the https://en.wikipedia.org/wiki/IBM_Z mainframe platform (identified as `s390x` in Python) isn't completely dead yet :slightly_smiling_face:
We haven't seen much interest in it, but we are nevertheless running normal tests and wheels on it since it is the only true big-endian architecture left in the wild, which is useful to verify that our C/Cython code doesn't make any endian-specific assumptions.

But anyway, let's limit `musllinux` support to x86 and ARM (both 64bit only).

**Comment by vytas7** [comment]
We will address this for Falcon 4.0 automatically courtesy of Cibuildwheel: #2259.

**Comment by vytas7** [comment]
Hi again @mwheeler! 
we will release Falcon wheels for `musllinux` as part of Falcon 4.0.0 release (about in one month time from now).

I have built out test wheels to TestPYPI, and they seem to work for me, but you can also take 'em for a spin in your container or `musllinux` distribution on metal: https://test.pypi.org/project/falcon/#files.

My limited testing:
`$ docker run --rm -it python:3.12-alpine /bin/sh`
```
/ # export PIP_INDEX_URL=https://test.pypi.org/simple/
/ # python -m venv /tmp/venv
/ # source /tmp/venv/bin/activate
(venv) / # pip install falcon
Looking in indexes: https://test.pypi.org/simple/
Collecting falcon
  Downloading https://test-files.pythonhosted.org/packages/0e/05/c196e5edcbee5107e68c9058314e17071db67f67d0063b19cbb9cf421aa9/falcon-4.0.0.dev1-cp312-cp312-musllinux_1_2_x86_64.whl.metadata (37 kB)
Downloading https://test-files.pythonhosted.org/packages/0e/05/c196e5edcbee5107e68c9058314e17071db67f67d0063b19cbb9cf421aa9/falcon-4.0.0.dev1-cp312-cp312-musllinux_1_2_x86_64.whl (11.9 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 11.9/11.9 MB 4.1 MB/s eta 0:00:00
Installing collected packages: falcon
Successfully installed falcon-4.0.0.dev1
(venv) / # python
Python 3.12.5 (main, Aug  7 2024, 19:15:55) [GCC 13.2.1 20240309] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import falcon
>>> falcon.parse_header
<cyfunction parse_header at 0x753d012312f0>
>>> falcon.parse_header('application/falcon; kind=peregrine')
('application/falcon', {'kind': 'peregrine'})
```

**Comment by vytas7** [comment]
@mwheeler: pre-built `musllinux` wheels are now [shipped as part of Falcon 4.0.1 (and 4.0.0)](https://falcon.readthedocs.io/en/stable/user/install.html#binary-wheels).

**Comment by mwheeler** [comment]
> @mwheeler: pre-built `musllinux` wheels are now [shipped as part of Falcon 4.0.1 (and 4.0.0)](https://falcon.readthedocs.io/en/stable/user/install.html#binary-wheels).

Thank you so much for this @vytas7 ! :)

I can't link the benefit directly (repo is private), but this brings our github action for building our service from 36m 32s down to 5m 20s :)

so over 30 minutes saved being able to use the musl wheels for aarch64 instead of building from source! 🎉   I'm not sure how many alpine (or other musl distros) based builds there out there using falcon on aarch64, but I'm sure we can't be the only ones... which means this is probably saving quite a lot of energy & time repeatedly building Falcon over and over and over (by default / if not using custom prebuilt wheels)
